### ISSUE TITLE
add 2 rpc calls to check if valid & registered

### ISSUE THREAD

**Issue by MSFTserver** [RavenProject/Ravencoin#411]
this is a feature request to add 2 new rpc calls the first one would be to check of an asset name is valid the second one would check if the asset name available of the name is not available return details about the asset incase someone is lookong to purchase that name, we could also combine these unless it really matters that they are seperate 
1. isValidAsset("AssetName")
2. isAvlAsset("AssetName")

_Sent from my Galaxy Note5 using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by MSFTserver** [comment]
i dont think you need ipfs to check if it exists im more interested in the address and seeing how messaging develops to contact the token owner

**Comment by russkidooski** [comment]
> return details about the asset incase someone is looking to purchase that name

contact data is stored in ipfs. eg. email address, domain name, contact name, if the asset is for sale, etc.

**Comment by bedri** [comment]
I think checking if asset exists or not and returning the IPFS hash with
other details as issuer pubkey etc. instead of full details about the asset
will work.

**Comment by MSFTserver** [comment]
we could setup a generalized ipsf sort of config for gwberal use to be adopted to list those details in the form of a "Contract", would be easier to parse out those details if people.follow a standard like json

**Comment by MSFTserver** [comment]
nevermind on my last comment, we have that (https://github.com/RavenProject/Ravencoin/blob/master/assets/asset_metadata_spec.md) but for the decoding of ipfs, i think it would be key to implement an option to enable ipfs, if people want to keep the wallet space low we could also have feature to ask with installer if they would want ipfs in addition to the wallet for asset metadata

**Comment by bedri** [comment]
I really like to see an **_optional_** IPFS installation along with Ravencoin installation. Of course no one should be forced to install IPFS but an option about it would be great.
So as a midway solution maybe we can first implement the suggested functions without IPFS optional installation requirement but then in time we can discuss and implement the optional installation of IPFS option.
As a primitive case we can do as I mentioned above (only the information that can be given right now such as IPFS hash, issuer pubkey, reissueable statement of the asset etc.) and then we can add the rest of the details when we add IPFS as an optionably installable entity to Ravencoin.
I think more discussion and opinions needed on adding IPFS to the default installation process even if it be optional.

**Comment by MSFTserver** [comment]
@bedri ill open another issue for ipfs install integration

**Comment by bedri** [comment]
That will be great

**Comment by MSFTserver** [comment]
#439
